aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -526,4 +526,57 @@ SparseTensor sparse_mask_cpu(const Tensor& t, const SparseTensor& mask) {
   return r;
 }
 
+Tensor sparse_mask_helper_cpu(
+  const SparseTensor& t,
+  const Tensor& mask_indices
+) {
+  /*
+    This is a helper function which filter values from `t._values()` using the `mask_indices`.
+    This CPU implementation uses a simple hash_map to filter values by matching the `mask_indices`
+    with the indices at tensor input `t`.
+
+    Inputs:
+      `t`             - tensor input
+      `mask_indices`  - mask indices tensor
+
+    Note: every index exists in the sparse tensor `t`
+  */
+  int64_t r_nnz = mask_indices.size(1);
```

Comment:
Does this assume that both tensors are coalesced?

Reply:
This assumes that only input parameter `t` is coalesced. I added this note to the documentation